podcasts/AppDelegate.swift

```diff
@@ -177,6 +177,13 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
                 return
             }
 
+            #if !os(watchOS)
+            if let timestamp {
```

Comment:
This has the possibility of saving the episode and the `playedUpTo` even if the user doesn't hit play.

I don't think it's a big issue but we can solve it by sharing the `timestamp` with the episode screen and only saving `playedUpTo` if the user hits play.

You can do that by changing `NavigationManager.sharedManager.navigateTo(NavigationManager.episodePageKey, data: [NavigationManager.episodeUuidKey: episode.uuid])` to `NavigationManager.sharedManager.navigateTo(NavigationManager.episodePageKey, data: [NavigationManager.episodeUuidKey: episode.uuid, "timestamp": timestamp])`

Then on `EpisodeDetailViewController` you can handle that. As a plus, we remove the `DataManager` from here.

What do you think?

Reply:
@leandroalonso I update the code with the suggestion above, the code is a bit more complex because there is more changes to get the timestamp information to the episode detail view. What do you think?